<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <group ns="camera">
        <node pkg="tj2_camera" type="tj2_imu_merger" name="camera_imu_merge">
            <param name="imu_base_frame" value="camera_link"/>
            <param name="publish_rate" value="15.0"/>
            <param name="sync_time_threshold" value="0.25"/>
            <!-- <remap from="imu" to="imu"/> -->
            <!-- <remap from="imu/filtered" to="imu/filtered"/> -->
            <remap from="joint" to="/tj2/camera_joint"/>
            <remap from="accel" to="accel/sample"/>
            <remap from="gyro" to="gyro/sample"/>
        </node>
        
        <node pkg="imu_filter_madgwick" type="imu_filter_node" name="camera_imu_filter">
            <param name="publish_tf" value="false"/>
            <param name="world_frame" value="enu"/>
            <param name="fixed_frame" value="camera_tilt_base_link"/>
            <param name="use_mag" value="false"/>
            <param name="use_magnetic_field_msg" value="false"/>
            <remap from="imu/data_raw" to="imu"/>
            <!-- <remap from="imu/data_raw" to="accel/sample"/> -->
            <remap from="imu/data" to="imu/filtered"/>
        </node>
    </group>
</launch>
